app/models/patient.rb

```diff
@@ -142,6 +166,24 @@ def self.contacted_since(datetime)
       first_contacts: first_contact.length, pledges_sent: 20 }
   end
 
+  def self.to_csv
+    CSV.generate(encoding: 'utf-8') do |csv|
+      csv << CSV_EXPORT_FIELDS.keys # Header line
+      all.each do |patient|
+        csv << CSV_EXPORT_FIELDS.values.map do |field|
+          value = patient.public_send(field)
```

Comment:
If I understand this correctly, we're checking the class of the field (e.g. string? array? int? decimal?) and if it's an array, stripping out blanks and then joining with commas and a space. That is pretty perfect!

I wonder if it's possible to write this just a tiny bit clearer -- would `value = patient[field.to_sym].class` fly here? I'll try that out and if it doesn't work out of the box we can roll with the `public_send`.